src/lasdi/latent_space.py

```diff
@@ -1,175 +1,482 @@
-            self.fcs += [torch.nn.Linear(layer_sizes[k], layer_sizes[k + 1])]
-        self.fcs = torch.nn.ModuleList(self.fcs)
-        self.init_weight()
+        -------------------------------------------------------------------------------------------
+        Arguments
+        -------------------------------------------------------------------------------------------
+
+        layer_sizes: A list of integers specifying the widths of the layers (including the 
+        dimensionality of the domain of each layer, as well as the co-domain of the final layer).
+        Suppose this list has N elements. Then the network will have N - 1 layers. The i'th layer 
+        maps from \mathbb{R}^{layer_sizes[i]} to \mathbb{R}^{layers_sizes[i]}. Thus, the i'th 
+        element of this list represents the domain of the i'th layer AND the co-domain of the 
+        i-1'th layer.
+
+        act_type: A string specifying which activation function we want to use at the end of each 
+        layer (except the final one). We use the same activation for each layer. 
+
+        reshape_index: This argument specifies if we should reshape the network's input or output 
+        (or neither). If the user specifies reshape_index, then it must be either 0 or -1. Further, 
+        in this case, they must also specify reshape_shape (you need to specify both together). If
+        it is 0, then reshape_shape specifies how we reshape the input before passing it through 
+        the network (the input to the first layer). If reshape_index is -1, then reshape_shape 
+        specifies how we reshape the network output before returning it (the output to the last 
+        layer). 
+
+        reshape_shape: These are lists of k integers specifying the final k dimensions of the shape
+        of the input to the first layer (if reshape_index == 0) or the output of the last layer 
+        (if reshape_index == -1). You must specify this argument if and only if you specify 
+        reshape_index. 
 
-        # Reshape input or output layer
+        threshold: You should only specify this argument if you are using the "Threshold" 
+        activation type. In this case, it specifies the threshold value for that activation (if x
+        is the input and x > threshold, then the function returns x. Otherwise, it returns the 
+        value).
+
+        value: You should only specify this argument if you are using the "Threshold" activation
+        type. In this case, "value" specifies the value that the function returns if the input is
+        below the threshold. 
+
+
+        -------------------------------------------------------------------------------------------
+        Returns
+        -------------------------------------------------------------------------------------------
+        
+        Nothing!
+        """
+
+        # Run checks.
+        # Make sure the reshape index is either 0 (input to 1st layer) or -1 (output of last 
```

Comment:
I don't think we need this long explanation for two lines 172-173. Line 160-163 is simply repeating what the code is doing. I suggest:

```
If the reshape index and shape are provided for input (index 0) or output (index -1),
we will either flatten the input ndarray or reshape the output 1d array into the provided ndarray shape.
Either way, the specified ndarray shape (reshape_shape) should match the size of input/output layer.
```